.github/workflows/cmake.yml

```diff
@@ -0,0 +1,966 @@
+on:
+- push
+- pull_request
+- pull_request_target
```

Comment:
Since this is a generated file, it would be a good idea to print a comment at the top declaring it so, and pointing people back at `scripts/ci_generator.py` to avoid wasting anyone's time if they're modifying this.

Reply:
yes, had it on my todo but got stuck in yaml land, will iron this out in some way.

Reply:
fixed.